src/lib/operators/table_scan/sorted_segment_search.hpp

```diff
@@ -0,0 +1,156 @@
+#pragma once
+
+#include <boost/range.hpp>
+#include <boost/range/join.hpp>
+#include <type_traits>
+
+#include "all_type_variant.hpp"
+#include "constant_mappings.hpp"
+#include "types.hpp"
+
+namespace opossum {
+
+namespace detail {
+
+// Generic class which handles the actual scanning of a sorted segment
+template <typename IteratorType, typename SearchValueType>
+class SortedSegmentSearch {
+ public:
+  SortedSegmentSearch(IteratorType begin, IteratorType end, const OrderByMode& order_by,
+                      const PredicateCondition& predicate_condition, const SearchValueType& search_value)
+      : _begin{begin},
+        _end{end},
+        _predicate_condition{predicate_condition},
+        _search_value{search_value},
+        _is_ascending{order_by == OrderByMode::Ascending || order_by == OrderByMode::AscendingNullsLast},
+        _is_nulls_first{order_by == OrderByMode::Ascending || order_by == OrderByMode::Descending} {}
+
+ private:
+  /**
+   * _get_first_bound and _get_last_bound are used to retrieve the lower and upper bound in a sorted segment but are
```

Comment:
If they are so analog to lower and upper bound, maybe call the "get_lower/upper_bound"?

Reply:
They were called like this initially but together with @klauck we decided to rename them because they return an offset and this might make things more clear.

Reply:
> They were called like this initially but together with @klauck we decided to rename them because they return an offset and this might make things more clear.

They don't return offsets any longer ;) 
I'm not opposed to renaming them.